Mapping/Loader/DoctrineIdentifierLoader.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the DunglasApiBundle package.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Dunglas\ApiBundle\Mapping\Loader;
+
+use Doctrine\Common\Annotations\Reader;
+use Dunglas\ApiBundle\Mapping\ClassMetadata;
+use Dunglas\ApiBundle\Util\ReflectionTrait;
+
+/**
+ * Doctrine identifier loader.
+ *
+ * @author Mikaël Labrut <labrut@gmail.com>
+ */
+class DoctrineIdentifierLoader implements LoaderInterface
+{
+    use ReflectionTrait;
+
+    /**
+     * @var string
+     */
+    const IDENTIFIER_ANNOTATION_NAME = 'Doctrine\ORM\Mapping\Id';
+
+    /**
+     * @var Reader
+     */
+    private $reader;
+
+    public function __construct(Reader $reader)
+    {
+        $this->reader = $reader;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function loadClassMetadata(
+        ClassMetadata $classMetadata,
+        array $normalizationGroups = null,
+        array $denormalizationGroups = null,
+        array $validationGroups = null
+    ) {
+        $reflectionClass = $classMetadata->getReflectionClass();
+
+        // If already set ignore
+        if ($classMetadata->getIdentifier()) {
+            return;
+        }
+
+        $identifier = null;
+        foreach ($reflectionClass->getProperties() as $reflectionProperty) {
+            if ($annotation = $this->reader->getPropertyAnnotation($reflectionProperty, self::IDENTIFIER_ANNOTATION_NAME)) {
```

Comment:
The annotation can be used without Doctrine so must be extracted from the doctrine specific loader (see @sroze comment).

Btw, we plan to add XML and YAML mappings format in future version, so the design should keep with the one of the `@Iri` loader: move the annotation specific part in the `AnnotationLoader`.